packages/net/writebuffer.pony

```diff
@@ -0,0 +1,69 @@
+use "collections"
+
+class WriteBuffer
+  """
+  A simple constant sized byte buffer to push Numbers into.
+  The data container is iso in order to keep it sendable(?)
+  Potential upgrades:
+  1. Have a list of data buffers of preset size
+  2. Arguably, seperate functions for _be and _le
+  3. A better way to get data out of the buffer, without passing a new array?
+  """
+  var data: Array[U8] iso
+  var _offset: USize
+  var _little_endian: Bool
+
+  new create(size: USize, little_endian: Bool = false) =>
+    """
+    Allocate the buffer
+    """
+    data = recover Array[U8].undefined(size) end
+    _offset = 0
+    _little_endian = little_endian
+
+  fun ref get_data(x: Array[U8] iso): Array[U8] iso^ =>
+    """
+    Destructive read inorder to get an iso reference to data
+    """
+    var x': Array[U8] iso = data = consume x
+    consume x'
+
+  fun ref available(): USize =>
+    """
+    Number of bytes currently available
+    """
+    data.size() - _offset
+
+  fun ref _byte(value: U8) : WriteBuffer^ ? =>
+    """
+    Write a single byte into the buffer if space available else error
+    Update _offset
+    """
+    if _offset < data.size() then
+      data(_offset) = value
+      _offset = _offset + 1
+    else error end
+    this
+
+  fun ref write_number(value: Number): WriteBuffer? =>
```

Comment:
I agree with most of your comments @jemc. Just out of curiosity, is there ever a time when both le and be methods are used? I still like the idea of setting the le/be at instantiation, and then unifying the call to
```pony
wb.u32(88)
```
where
```pony
fun ref u32 (val: U32) =>
  if (_little_endian) u32_le(val) else u32_be(val) end
```
Unfortunately, it's an extra function call, and also moves away from Buffer.